docs/src/views/IntegrateCustomComponents.tsx

```diff
@@ -0,0 +1,239 @@
+      your custom component - e.g. by simply passing them as props. This <code>render</code> method
+      will be invoked with the following parameters:
+    </p>
+    <ul>
+      <li>
+        <code>stardust</code> - the object containing the evaluated theming props (<code>
+          classes
+        </code>
+        and <code>rtl</code>).
+      </li>
+      <li>
+        <code>...props</code> - all other props provided by the user.
+      </li>
+    </ul>
+
+    <Header as="h2" content="Using the custom components" />
+
+    <p>
+      We already saw how the <code>Provider</code> can define some stylings and variables for the
+      custom components. Next, we will take a look into several examples of how the user can further
+      customize styles and variables of these components, the same way they would do with the
+      Stardust components.
+    </p>
+
+    <Header
+      as="h3"
+      content={
+        <>
+          Example 1. Using <code>styles</code> property
+        </>
+      }
+    />
+
+    <ExampleSnippet
+      value={[
+        `<StyledButton styles={{backgroundColor: "black"}}>`,
+        `  Inline styled button`,
+        `</StyledButton>`,
+      ].join('\n')}
+      render={() => (
+        <Provider
+          theme={{
+            componentVariables: {
+              StyledButton: {
+                color: '#F2F2F2',
+              },
+            },
+            componentStyles: {
+              StyledButton: {
+                root: ({ props, variables, theme: { siteVariables } }) => ({
+                  backgroundColor: siteVariables.brand,
+                  color: (variables as any).color,
+                }),
+              },
+            },
+          }}
+        >
+          <StyledButton styles={{ backgroundColor: 'black' }}>Inline styled button</StyledButton>
+        </Provider>
+      )}
+    />
+
+    <Header
+      as="h3"
+      content={
+        <>
+          Example 2. Using <code>variables</code> property
+        </>
+      }
+    />
+
+    <ExampleSnippet
+      value={[
+        `<StyledButton variables={{color: "black" }}>`,
+        `  Inline styled button`,
+        `</StyledButton>`,
+      ].join('\n')}
+      render={() => (
+        <Provider
+          theme={{
+            componentVariables: {
+              StyledButton: {
+                color: '#F2F2F2',
+              },
+            },
+            componentStyles: {
+              StyledButton: {
+                root: ({ props, variables, theme: { siteVariables } }) => ({
+                  backgroundColor: siteVariables.brand,
+                  color: (variables as any).color,
+                }),
+              },
+            },
+          }}
+        >
+          <StyledButton variables={{ color: 'black' }}>inline styled button</StyledButton>
+        </Provider>
+      )}
+    />
+
+    <p>
+      All other advanced scenarios for applying styles and variables are supported as well. For more
```

Comment:
Would suggest to write it this way: 

- for each example (Styles and Variables) introduce the concluding paragraph that would tell the following (essentially, the though expressed in this paragraph: 'Note that all other scenarios for applying styles/variables are supported as well - please, take a look on the corresponding ... section for more details.')

- introduce additional example where theming support will be mentioned (primitive example is necessary):

```jsx
const customTheme = {
  // component's displayName arg is used as a selector
  StyledButton: {
   ...
  }
}

<Provider theme={customTheme}>
  <StyledButton>
</Provider>
```

- and, the same way, conclude with a similar paragraph saying 'for more advanced theming scenarios, please, take a look on the corresponding Theming section'